plugins/gcp/resoto_plugin_gcp/resources/compute.py

```diff
@@ -3292,6 +3292,8 @@ def collect_individual(cls: Type[GcpResource], builder: GraphBuilder, zone: str,
             machineType=name,
         )
         machine_type_obj = GcpMachineType.from_api(result)
+        machine_type_obj._zone = builder.zone_by_name[zone]
+        machine_type_obj._region = builder.region_by_zone_name[zone]
```

Comment:
So we collect machine types for a specific zone. To reflect the client behaviour for zonal responses, can't we simply do:
```python
result[InternalZoneProp]=zone
```
This should take care for setting the relevant properties and adding edges.